### PR TITLE
Reinitialize potentially disconnected keepkeys, add disconnectAll method, attempt to open/close regardless of keepkey state

### PR DESCRIPTION
**Pull request by spiceboi**

### CODE REVIEW

File: src/device.ts [hunk 1]
```diff
@@ -45,7 +46,10 @@ export default abstract class Device {
     // If error, throw with response message
     if (responseTypeEnum === Messages.MessageType.MESSAGETYPE_FAILURE) {
       const errorResponse = responseMsg as Messages.Failure
-      throw new Error(errorResponse.getMessage())
+
+      if (errorResponse.getCode() !== 4) {
```
Comment (keepkeyjon) [inline src/device.ts]: Is there a protobuf enum we can use instead of the 4?

File: src/webUSBDevice.ts [hunk 2]
```diff
@@ -36,18 +36,15 @@ export default class WebUSBDevice extends Device {
...
+    if (this.usbDevice.configuration === null) await this.usbDevice.selectConfiguration(1)
+    await this.usbDevice.claimInterface(0)
   }
 
   public async disconnect (): Promise<void> {
-    if (!this.usbDevice.opened) return
     try {
       // If the device is disconnected, this will fail and throw, which is fine.
-      await this.usbDevice.releaseInterface(0)
+      await this.usbDevice.close()
```
Comment (majorhayes) [inline src/webUSBDevice.ts]: Just curious: Why is close better?

Comment (keepkeyjon) [reply src/webUSBDevice.ts]: shouldn't it be both?

Comment (spiceboi) [reply src/webUSBDevice.ts]: From mdn: The close() method of the USBDevice interface returns a promise that resolves when all open interfaces are released and the device session has ended.

Seems like close releases interfaces, in the case of browser tabs, we're trying to beat a race condition and I'd like to have fewer async calls in flight as possible so the next tab can hopefully reconnect.

Comment (keepkeyjon) [reply src/webUSBDevice.ts]: SGTM

**Pull request merged**